packages/data-api/src/AnalyticsFetchQuery.js

```diff
@@ -3,119 +3,130 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 
-import { DataFetchQuery, ANSWER_SPECIFIC_FIELDS } from './DataFetchQuery';
+import { DataFetchQuery } from './DataFetchQuery';
+
+const VALUE_AGGREGATION_FUNCTIONS = {
+  SUM: 'sum(value::numeric)::text',
+  MOST_RECENT: 'most_recent(value, date)',
+};
 
 const AGGREGATION_SWITCHES = {
   FINAL_EACH_DAY: {
     groupByPeriodField: 'day_period',
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
   },
   FINAL_EACH_WEEK: {
     groupByPeriodField: 'week_period',
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
+  },
+  SUM_EACH_WEEK: {
+    groupByPeriodField: 'week_period',
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.SUM,
   },
   FINAL_EACH_MONTH: {
     groupByPeriodField: 'month_period',
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
   },
   FINAL_EACH_YEAR: {
     groupByPeriodField: 'year_period',
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
   },
   MOST_RECENT: {
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
   },
   MOST_RECENT_PER_ORG_GROUP: {
-    getLatestPerPeriod: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.MOST_RECENT,
     aggregateEntities: true,
   },
   SUM_PER_ORG_GROUP: {
-    sum: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.SUM,
     aggregateEntities: true,
   },
   SUM_PER_PERIOD_PER_ORG_GROUP: {
-    sum: true,
+    aggregationFunction: VALUE_AGGREGATION_FUNCTIONS.SUM,
     aggregateEntities: true,
-    groupByPeriodField: 'day_period', // can assume first internal aggregation period type is daily
   },
 };
 
 export class AnalyticsFetchQuery extends DataFetchQuery {
   constructor(database, options) {
     super(database, options);
 
-    const firstAggregation = options.aggregations?.[0];
-    this.aggregation = { type: firstAggregation?.type };
-    this.aggregation.switches = AGGREGATION_SWITCHES[firstAggregation?.type]; // add internal switches
-    this.aggregation.config = firstAggregation?.config; // add external config, supplied by client
-    this.isAggregating = !!this.aggregation.switches;
+    this.aggregations = [];
+    const originalAggregations = [...options.aggregations];
+    for (let i = 0; i < originalAggregations.length; i++) {
+      const aggregation = originalAggregations[i];
+      if (!AGGREGATION_SWITCHES[aggregation?.type]) {
+        break; // We only support chaining aggregations if all aggregations are supported type
```

Comment:
This comment needs an update, should be `Only chain aggregations up until the first unsupported type`